### PR TITLE
cmd: add more test for cmd

### PR DESCRIPTION
**Pull request by zhijian-pro**
cmd: add more test for cmd

### CODE REVIEW

File: cmd/dump_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	})
+	t.Run("Test dump", func(t *testing.T) {
+		dumpArgs := []string{"", "dump", metaUrl, "/tmp/dump_test.json"}
+
+		Main(dumpArgs)
+		_, err := os.Stat("/tmp/dump_test.json")
+		if err != nil {
+			t.Fatalf("dump error: %v", err)
+		}
+	})
```
Comment (davies) [inline cmd/dump_test.go:51]: Add a case for subdir

File: cmd/main.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ func globalFlags() []cli.Flag {
 	}
 }
 
-func main() {
+func Main(args []string) {
```
Comment (davies) [inline cmd/main.go:59]: return error

File: cmd/main.go [hunk 3]
```diff
@@ -97,12 +97,16 @@ func main() {
 		}
 	}
 
-	err := app.Run(reorderOptions(app, os.Args))
+	err := app.Run(reorderOptions(app, args))
 	if err != nil {
 		log.Fatal(err)
```
Comment (davies) [inline cmd/main.go:102]: move the `log.Fatal(err)` into main()

File: cmd/info_test.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	for i := 0; i < 10; i++ {
+		filename := fmt.Sprintf("%s/dir1/f%d.txt", mountpoint, i)
+		err := ioutil.WriteFile(filename, []byte("test"), 0644)
+		if err != nil {
+			t.Fatalf("mount failed: %v", err)
+		}
+	}
+
+	infoArgs := []string{"", "info", metaUrl, fmt.Sprintf("%s/dir1", mountpoint)}
+	Main(infoArgs)
```
Comment (davies) [inline cmd/info_test.go:43]: check the output.

We replace os.Stdout to bytes writer before calling Main() to capture the output.

Comment (zhijian-pro) [reply cmd/info_test.go:43]: ok

File: cmd/umount_test.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		t.Fatalf("mount failed: %v", err)
+	}
+
+	err := UmountTmp(mountpoint)
+	if err != nil {
+		t.Fatalf("umount failed: %v", err)
+	}
+
+	t.Log("---------------")
+	println(prometheus.DefaultRegisterer)
```
Comment (davies) [inline cmd/umount_test.go:43]: ?

Comment (davies) [reply cmd/umount_test.go:43]: We should check that the /tmp/testDir is not a mountpoint

File: cmd/rmr_test.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+		t.Fatalf("rmr failed : %v", err)
+	}
+
+	for _, path := range paths {
+		dir, err := os.ReadDir(mountpoint + path)
+		if len(dir) != 0 {
+			t.Fatalf("test rmr error: %v", err)
+		}
+	}
+	err := UmountTmp(mountpoint)
```
Comment (davies) [inline cmd/rmr_test.go:58]: call it with defer

File: cmd/info.go [hunk 7]
```diff
@@ -49,6 +50,14 @@ func infoFlags() *cli.Command {
 	}
 }
 
+var writerInfo = func(content string, w io.Writer) error {
```
Comment (davies) [inline cmd/info.go:53]: We should not messy the source code just because of tests.

**Review by davies — APPROVED** [review]

**Pull request merged**